app/controllers/agreements_controller.rb

```diff
@@ -0,0 +1,13 @@
+class AgreementsController < ApplicationController
+  def index
+    tenancy_ref = agreements_params.fetch(:tenancy_ref)
+
+    response = Hackney::Income::ViewAgreements.execute(tenancy_ref: tenancy_ref)
```

Comment:
Can we call this via the `income_use_case_factory` like we do in the other controllers just for consistency sake? 
```suggestion
    response = income_use_case_factory.view_agreements.execute(tenancy_ref: tenancy_ref)
```